tools/codegen/generator.go

```diff
@@ -345,7 +346,11 @@ func generateParseNumArgsCode(definition eventDefinition) string {
 	return fmt.Sprintf(`  var numArgs uint8
   err = decoder.DecodeUint8(&numArgs)
 %s
-`, generateDecoderErrorCheck(definition))
+
+  if numArgs != %d {
+    log.Warnf("unexpected number of args received when parsing '%sArgs': wanted %d, got %%d", numArgs)
```

Comment:
Maybe is such case would be better to return error here instead of logging it. If there is numArgs mismatch event data is possible wrong any way.